massa-execution-worker/src/speculative_roll_state.rs

```diff
@@ -146,6 +146,37 @@ impl SpeculativeRollState {
             .entry(target_slot)
             .or_insert_with(PreHashMap::default);
         credit.insert(*seller_addr, roll_price.saturating_mul_u64(roll_count));
+        Ok(())
+    }
+
+    /// Try to burn `roll_count` rolls from the seller address.
+    ///
+    /// # Arguments
+    /// * `addr`: address to sell the rolls from
+    /// * `roll_count`: number of rolls to sell
+    pub fn try_burn_rolls(
+        &mut self,
+        addr: &Address,
+        roll_count: u64,
+    ) -> Result<(), ExecutionError> {
```

Comment:
this function should not fail if it can slash only part of the rolls, but return the number of rolls effectively slashed

Reply:
done